src/graphnet/data/i3extractor.py

```diff
@@ -380,27 +392,26 @@ def __call__(self, frame) -> dict:
             for classifier in classifiers:
                 if frame_has_key(frame, classifier):
                     output.update({classifier : frame[classifier].value})
-            #output.update({
-            #    'L7_MuonClassifier_FullSky_ProbNu': frame["L7_MuonClassifier_FullSky_ProbNu"].value,
-            #    'L4_MuonClassifier_Data_ProbNu': frame["L4_MuonClassifier_Data_ProbNu"].value,
-            #    'L4_NoiseClassifier_ProbNu': frame["L4_NoiseClassifier_ProbNu"].value,
-            #    'L7_PIDClassifier_FullSky_ProbTrack': frame["L7_PIDClassifier_FullSky_ProbTrack"].value,
-            #})
 
         if frame_is_montecarlo(frame):
-            if frame_contains_retro(frame):
+            if frame_is_noise(frame):
                 output.update({
-                    'osc_weight': frame["I3MCWeightDict"]["weight"],
+                    'osc_weight': frame["noise_weight"]["weight"],
                 })
```

Comment:
@RasmusOrsoe, does this change affect you? Perhaps it would be better to introduce a new output variable called `noise_weight` or similar?

Reply:
Hey @asogaard. This looks suspicious, so thanks for pointing it out! 

**However**, I think this is a change following an offline conversation I had with Leon. The `I3MCWeightDict` weight weighs events to physical rates. If I remember correctly, this weight is not stored in `["I3MCWeightDict"]` for pure noise events, and I think this is why Leon is adding it from a different field for noise events. @BozianuLeon can you confirm?

Reply:
I can confirm what you are saying @RasmusOrsoe, this is oscNext's way of handling weights of noise events, just storing it somewhere differently. I do not think we lose any information by overwriting osc_weight here